core/app/finders/spree/products/find.rb

```diff
@@ -263,7 +263,7 @@ def map_prices(prices)
       def taxon_ids(taxons_ids)
         return if taxons_ids.nil? || taxons_ids.to_s.blank?
 
-        taxons = store.taxons.where(id: taxons_ids.to_s.split(','))
+        taxons = Spree::Taxon.for_store(store).where(id: taxons_ids.to_s.split(','))
```

Comment:
@mbajur `store.taxons` and `Spree::Taxon.for_strore(store)` should give you the exact same records :)

Reply:
Yup! This PR is mostly about DRY and predictibility. I recently worked on a shop that has a requirement of all taxons being shared between all the Stores. In order to achieve that, i modified the `for_store` scope on `Spree::Taxon` to soon realize products listing is not filtering by current taxon because of it not relying on `for_store` but on hardcoded `store.taxons` call - therefore IMO having Finder using the same scope as rest of the app would make it a bit easier to work with such customization and to prevent user from unexpected outcome :)